arch/arm/arm64/memtag.c

```diff
@@ -72,11 +72,11 @@ int ukarch_memtag_init(void)
 #endif
 
 	/* Set seed for RGSR_EL1 */
-	if (ukarch_random_init())
-		UK_CRASH("Arch random not available\n");
-
-	if (ukarch_random_seed_u64(&seed))
-		UK_CRASH("Could not generate MTE seed\n");
+	rc = uk_random_fill_buffer(&seed, sizeof(seed));
+	if (unlikely(rc)) {
+		uk_pr_err("Could not generate MTE key (%d)\n", rc);
```

Comment:
> Somehow I am tending to use "Failed to generate..." instead of "Could not generate..." since it is underlining that there is a serious problem. However, we are talking here probably about preference.

Indeed, I think this is subjective :) Saying that, the idea is that the library does not enforce policy, so the severity of a failure is decided by the caller. In our case we treat failures of initcalls as critical.

> If we fail, what is the actual effect? Are we able to boot or do we fail if we do not have MTE?

Right now we call `UK_CRASH()`. The same will happen when we move MTE initialization to `earlytab`. IMO, if I had to choose between continuing boot with a warning or crashing, I would favor security and choose the latter, given that the fact that someone enabled MTE explicitly in Kconfig implies that they care about that feature. Nevertheless, we can also consider adding a feature in `libukboot` or individual libraries like `ukmemtag` that allows the user to select the system behavior upon failure.